src/neo/SmartContract/Native/Tokens/Nep11Token.cs

```diff
@@ -0,0 +1,225 @@
+                        }
+                    },
+                    ReturnType = ContractParameterType.Boolean
+                }
+            };
+
+            Manifest.Abi.Events = events.ToArray();
+        }
+
+        [ContractMethod(0_01000000, CallFlags.AllowStates)]
+        public virtual BigInteger TotalSupply(StoreView snapshot)
+        {
+            StorageItem storage = snapshot.Storages.TryGet(CreateStorageKey(Prefix_TotalSupply));
+            if (storage is null) return BigInteger.Zero;
+            return new BigInteger(storage.Value);
+        }
+
+        [ContractMethod(0_01000000, CallFlags.AllowStates)]
+        public virtual BigInteger BalanceOf(StoreView snapshot, UInt160 account, byte[] tokenId)
+        {
+            if (tokenId.Length > MaxTokenIdLength) return BigInteger.Zero;
+            UInt160 innerKey = GetInnerKey(tokenId);
+            StorageItem storage = snapshot.Storages.TryGet(CreateOwnershipKey(Prefix_OwnerToTokenId, account, innerKey));
+            if (storage is null) return BigInteger.Zero;
+            return storage.GetInteroperable<TAccount>().Balance;
+        }
+
+        [ContractMethod(0_01000000, CallFlags.AllowStates)]
+        public virtual IEnumerator TokensOf(StoreView snapshot, UInt160 owner)
+        {
+            return new CollectionWrapper(snapshot.Storages.Find(CreateStorageKey(Prefix_OwnerToTokenId, owner).ToArray()).Select(p =>
+            {
+                UInt160 innerKey = new UInt160(p.Key.Key.Skip(1 + UInt160.Length).Take(UInt160.Length).ToArray());
+                TToken token = snapshot.Storages.TryGet(CreateTokenKey(innerKey)).GetInteroperable<TToken>();
+                return (StackItem)token.Id;
+            }).GetEnumerator());
+        }
+
+        [ContractMethod(0_01000000, CallFlags.None)]
+        public IEnumerator OwnerOf(ApplicationEngine engine, byte[] tokenId)
+        {
+            if (tokenId.Length > MaxTokenIdLength) throw new InvalidOperationException("Invalid tokenId");
+
+            var owner = OwnerOf(engine.Snapshot, tokenId);
+            if (owner is null) throw new InvalidOperationException("Token doesn't exist");
+
+            var array = new ArrayWrapper(new StackItem[] { owner.ToArray() });
+            return array;
+        }
+
+        public virtual UInt160 OwnerOf(StoreView snapshot, byte[] tokenId)
+        {
+            UInt160 innerKey = GetInnerKey(tokenId);
+            return snapshot.Storages.Find(CreateStorageKey(Prefix_TokenIdToOwner, innerKey).ToArray()).Select(p =>
+            {
+                var owner = p.Key.Key.Skip(1 + UInt160.Length).Take(UInt160.Length).ToArray();
+                return new UInt160(owner);
+            })?.First();
+        }
+
+        [ContractMethod(0_08000000, CallFlags.AllowModifyStates)]
+        public virtual bool Transfer(ApplicationEngine engine, UInt160 to, byte[] tokenId)
+        {
+            if (tokenId.Length > MaxTokenIdLength) return false;
```

Comment:
Accorging to 
https://github.com/neo-project/proposals/pull/41/files?short_path=afd2225#diff-afd22253b102da74b022d8c022201ee6

`The parameter tokenid SHOULD be a valid NFT. If not, this method SHOULD throw an exception.`

Reply:
So how to define a valid NFT ID? I think `MaxTokenIdLength` is enough.